docs/architecture/distributed-cloud-native-apps-containers/api-gateways/gateway-patterns.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Gateway patterns
+description: Architecture for Distributed Cloud-Native Apps with .NET Aspire & Containers | Gateway patterns
+author: 
+ms.date: 04/25/2024
+---
+
+# Gateway patterns
+
+[!INCLUDE [download-alert](../includes/download-alert.md)]
+
+!TODO - sorry couldn't get these done in time. Should be easy to refresh from my powerpoint.
+
+In a cloud-native system, front-end clients (mobile, web, and desktop applications) require a communication channel to interact with independent back-end microservices.
+
+What are the options?
+
+To keep things simple, a front-end client could *directly communicate* with the back-end microservices.
+
+![Direct client to service communication](./media/direct-client-to-service-communication.png)
+
+**Figure 13-1**. Direct client to service communication
+
+With this approach, each microservice has a public endpoint that is accessible by front-end clients. In a production environment, you'd place a load balancer in front of the microservices, routing traffic proportionately.
+
+While simple to implement, direct client communication would be acceptable only for simple microservice applications. This pattern tightly couples front-end clients to core back-end services, opening the door for many problems, including:
+
+- Client susceptibility to back-end service refactoring.
+- A wider attack surface as core back-end services are directly exposed.
+- Duplication of cross-cutting concerns across each microservice.
+- Overly complex client code - clients must keep track of multiple endpoints and handle failures in a resilient way.
+
+Instead, a widely accepted cloud design pattern is to implement an [API Gateway Service](../microservices/architect-microservice-container-applications/direct-client-to-microservice-communication-versus-the-api-gateway-pattern.md) between the front-end applications and back-end services.
```

Comment:
This file doesnt exists ../microservices/architect-microservice-container-applications/direct-client-to-microservice-communication-versus-the-api-gateway-pattern.md ?

Reply:
Fixed. Needed to go two levels up.